### PR TITLE
ref(discover): Use new Overlay for Save as dropdown

### PR DESCRIPTION
**Pull request by vuluongj20**
**Before:**
Screen Shot 2022-07-06 at 6 12 06 PM

**After:**
Screen Shot 2022-07-06 at 6 12 25 PM

### CODE REVIEW

File: static/app/views/eventsV2/savedQuery/index.tsx [hunk 1]
```diff
@@ -12,22 +15,74 @@ import Banner from 'sentry/components/banner';
...
+  onChangeInput,
+  modifiedHandleCreateQuery,
+}: SaveAsDropdownProps) {
+  const {isOpen, triggerProps, overlayProps, arrowProps} = useOverlay({});
+  const theme = useTheme();
+
+  return (
+    <div>
+      <Button icon={<IconStar />} aria-disabled={disabled} {...triggerProps}>
+        Save as
```
Comment (priscilawebdev) [inline static/app/views/eventsV2/savedQuery/index.tsx:54]:
```suggestion
       {t('Save as')}
```

File: static/app/views/eventsV2/savedQuery/index.tsx [hunk 2]
```diff
@@ -12,22 +15,74 @@ import Banner from 'sentry/components/banner';
...
+  queryName: string;
+};
+
+function SaveAsDropdown({
+  queryName,
+  disabled,
+  onChangeInput,
+  modifiedHandleCreateQuery,
+}: SaveAsDropdownProps) {
+  const {isOpen, triggerProps, overlayProps, arrowProps} = useOverlay({});
```
Comment (priscilawebdev) [inline static/app/views/eventsV2/savedQuery/index.tsx:48]: should we update the `useOverlay` so then we don't need to pass an empty object in this case?

File: static/app/views/eventsV2/savedQuery/index.tsx [hunk 3]
```diff
@@ -12,22 +15,74 @@ import Banner from 'sentry/components/banner';
...
+                  placeholder={t('Display name')}
+                  value={queryName || ''}
+                  onChange={onChangeInput}
+                  disabled={disabled}
+                />
+                <Button
+                  onClick={modifiedHandleCreateQuery}
+                  priority="primary"
+                  disabled={disabled || !queryName}
+                  style={{width: '100%'}}
```
Comment (priscilawebdev) [inline static/app/views/eventsV2/savedQuery/index.tsx:73]: should we use emotion here to style the button?

File: static/app/views/eventsV2/savedQuery/index.tsx [hunk 4]
```diff
@@ -12,22 +15,74 @@ import Banner from 'sentry/components/banner';
...
+      <Button icon={<IconStar />} aria-disabled={disabled} {...triggerProps}>
+        Save as
+      </Button>
+      <AnimatePresence>
+        {isOpen && (
+          <FocusScope contain restoreFocus autoFocus>
+            <PositionWrapper zIndex={theme.zIndex.dropdown} {...overlayProps}>
+              <StyledOverlay arrowProps={arrowProps} animated>
+                <ButtonSaveInput
+                  type="text"
```
Comment (priscilawebdev) [inline static/app/views/eventsV2/savedQuery/index.tsx:62]: it seems a bit odd that we define the type "text" for a "Button". Is there no other name we can use for this component?

File: static/app/views/eventsV2/savedQuery/index.tsx (hunk 3 above)
Comment (vuluongj20) [reply static/app/views/eventsV2/savedQuery/index.tsx]: True, updated in https://github.com/getsentry/sentry/pull/36402/commits/81bdc9ee8fb10a631a4e1166d7e76a94dd6bfd20

**Comment by priscilawebdev** [comment]
maybe something for a follow-up, but after saving, the star is almost invisible 

image

**Review by priscilawebdev — APPROVED** [review]
I left a few comments in the tests, but it looks good to me

File: tests/js/spec/views/eventsV2/savedQuery/index.spec.tsx [hunk 5]
```diff
@@ -223,24 +205,15 @@ describe('EventsV2 > SaveQueryButtonGroup', function () {
...
-      const buttonSaveAs = wrapper.find('DropdownControl').first();
-      buttonSaveAs.simulate('click');
+      userEvent.click(await screen.findByRole('button', {name: 'Save as'}));
 
       // Do not fill in Input
-      await tick();
 
       // Click on Save in the Dropdown
-      buttonSaveAs.find('ButtonSaveDropDown Button').simulate('click');
+      userEvent.click(await screen.findByRole('button', {name: 'Save for Org'}));
```
[the remark this answers is not in the document]
Comment (vuluongj20) [reply tests/js/spec/views/eventsV2/savedQuery/index.spec.tsx]: Turns out all of these awaits can be deleted – good catch! – removed them in [ref(discover-tests): Remove unnecessary awaits](https://github.com/getsentry/sentry/pull/36402/commits/7c905795046a41e2c7a87a33fa81da67304dfc48)

**Pull request merged**